pallets/creditcoin/src/tests.rs

```diff
@@ -3236,3 +3237,47 @@ fn register_address_v2_should_error_with_unsupported_blockchain() {
 		);
 	});
 }
+
+use hex_literal::hex;
+
+#[test]
+fn set_burn_gate_contract_should_return_default_goerli_contract_when_not_set() {
+	ExtBuilder::default().build_and_execute(|| {
+		let contract: DeployedContract = Creditcoin::gate_contract();
+
+		assert_eq!(
+			contract.address,
+			sp_core::H160(hex!("a3EE21C306A700E682AbCdfe9BaA6A08F3820419"))
+		);
+
+		assert_eq!(contract.chain, Blockchain::Ethereum);
+	});
+}
+
+#[test]
+fn set_burn_gate_contract_fails_with_non_root() {
+	ExtBuilder::default().build_and_execute(|| {
+		let acct: AccountId = AccountId::new([0; 32]);
+		let gate_contract = DeployedContract::default();
+
+		assert_noop!(Creditcoin::set_gate_contract(Origin::signed(acct), gate_contract), BadOrigin);
+	});
+}
+
+#[test]
+fn set_burn_gate_contract_passes_and_storage_is_updated() {
```

Comment:
```suggestion
fn set_gate_contract_passes_and_storage_is_updated() {
```

- prefix with the actual extrinsic name to make tests easier to find